lib/dpl/provider/elastic_beanstalk.rb

```diff
@@ -88,7 +92,13 @@ def create_zip
       end
 
       def upload(key, file)
-        obj = s3.buckets[bucket_name].objects[key]
+        obj = s3.buckets[bucket_name]
+        file_path = bucket_path
+        if file_path
+          file_path << '/' unless file_path.end_with?('/')
```

Comment:
This combination of `if` and `unless` doesn't read well.

`file_path.gsub!(/\/*$/,'/')` should be sufficient.

Also, what is the reason for `file_path = bucket_path`? I don't think `bucket_path` is used elsewhere, and just having `file_path` pointing to the same object as `bucket_path` does not make sense to me.